apps/tasks.py

```diff
@@ -148,6 +179,39 @@ def deploy_resource(serialized_instance):
     instance.app_status.save()
     instance.save()
 
+    # In development, also generate and validate the k8s deployment manifest
+    if settings.DEBUG:
```

Comment:
I think this whole code block deserves to be a separate procedure, I think it would help with the readability of the function

Reply:
I agree. I am just starting work on this PR and it is still a draft so there it will continue to change a lot before it is ready.